yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -197,6 +197,9 @@ private[spark] class ApplicationMaster(
         System.setProperty("spark.yarn.app.id", appAttemptId.getApplicationId().toString())
       }
 
+      val context = s"${System.getProperty("spark.app.name")} running on Spark"
```

Comment:
application name can be gotten from SparkConf, also I think you should take care of app name not set issue, here may probably throw exception.